src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -147,6 +149,7 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
             }
 
             _logger = logger;
+            _vulnerabilitiesFoundService = vulnerabilitiesFoundService;
```

Comment:
I don't this was a good change. 
You want to avoid depending on null checks in scenarios where we're always going to have the service. 
We declare the service, if there's an issue getting the service, then we have a problem with our code. That code shouldn't be ignored or considered expected. 

I'm not sure there's any failures in MEF that are non breaking.

Reply:
I specifically suggested a fault telemetry, not ignoring it. However, if MEF is expected to be breaking when it fails on ANY service (to reiterate: this doesn't seem like a critical service to me), I agree, leave the exception.

Reply:
> I specifically suggested a fault telemetry, not ignoring 

I'm not suggesting you did. But the implementation right now does that.

Badly configured MEF is similar as a badly configured nuget.config.